58 datacite badge

amadulhaxxani opened the pull request:
- Adding DataCite badge in simple item view
- conditionally rendering the badge if appConfig.datacite has value
- loading the scripts dynamically only when appConfig.datacite has value
- tests included

amadulhaxxani committed:
adding config for datacite

adding config for datacite

amadulhaxxani committed:
Added a new component for datacite badge

Added a new component for datacite badge

amadulhaxxani committed:
translations for datacite

translations for datacite

amadulhaxxani committed:
adding datacite in untyped item

adding datacite in untyped item

amadulhaxxani committed:
lint fix

lint fix

amadulhaxxani committed:
added basic tests for testing initial component

added basic tests for testing initial component

amadulhaxxani committed:
tests for item handling

tests for item handling

amadulhaxxani committed:
testing extraction of DOI

testing extraction of DOI

amadulhaxxani committed:
adding tests for badge visibility

adding tests for badge visibility

amadulhaxxani committed:
lint fix

lint fix

amadulhaxxani committed:
making fields public

making fields public

amadulhaxxani commented on src/app/item-page/datacite/datacite-badge.component.spec.ts:
```diff
@@ -0,0 +1,484 @@
+import { MetadataMap, MetadataValue } from 'src/app/core/shared/metadata.models';
+import { Data } from '@angular/router';
+import { NO_ERRORS_SCHEMA, PLATFORM_ID } from '@angular/core';
+import exp from 'constants';
```
removed with lint fix

amadulhaxxani committed:
copilot fixes

copilot fixes

kosarko commented on src/app/item-page/datacite/datacite-badge.component.ts:
```diff
@@ -0,0 +1,141 @@
+    try {
+      await this.loadScript('https://unpkg.com/vue@^2/dist/vue.min.js');
+      await this.loadScript('https://unpkg.com/@webcomponents/webcomponentsjs@2.0.0/webcomponents-loader.js');
+      await this.loadScript('https://unpkg.com/data-metrics-badge/dist/data-metrics-badge.min.js');
```
Any benefits to loading the scripts this way? Why not have it as a dependency in package.json?

amadulhaxxani replied:
I don't think that we had the package available
or maybe I missed it somewhere...
you mean installing the package with npm yes?

amadulhaxxani replied:
Okay you are right 
I don't know why I just relied on that one link related to datacite
I will implement it using the dependency in package.json :(

amadulhaxxani replied:
I tried implementing it using the npm package but it's somehow not possible. Maybe some engineering skills will be required but I am not sure if we need to waste this much time or not. I already wasted some in trying to find some alternative or making it work
Some nice reason from copilot for this approach are below:
- The badge depends on Vue 2 and a Web Components polyfill; importing it via the Angular build (SSR-enabled) triggers “window/document” usage on the server and breaks Angular Universal. Loading scripts in ngAfterViewInit under isPlatformBrowser avoids SSR errors.
- The package is published primarily as a UMD/minified bundle for <script> usage, not a clean, tree-shakeable ESM module with typings; bundling it directly bloats the main bundle and complicates TypeScript/Angular builds.
- Script order matters (Vue → webcomponents-loader → data-metrics-badge). Loading them explicitly ensures correct sequencing without wrestling with the Angular/webpack pipeline.
- This approach keeps our app bundle lean and defers loading until a DOI is present and the badge is shown, improving performance.
- It also isolates potential version conflicts (Vue 2 in an Angular app) and avoids polluting the global build

kosarko commented on src/config/app-config.interface.ts:
```diff
@@ -51,6 +51,7 @@ interface AppConfig extends Config {
   comcolSelectionSort: DiscoverySortConfig;
   signpostingEnabled: boolean;
   matomo: MatomoConfig;
+  datacite?: string;
```
`datacite` feels as a too generic name, maybe `dataciteBadgeField` ?

amadulhaxxani committed:
Rename datacite config field to dataciteBadgeField

Updated AppConfig interface and related component code to use 'dataciteBadgeField' instead of 'datacite' for clarity and consistency in Datacite badge configuration.

amadulhaxxani committed:
Merge branch 'clarin-v7' into 58-datacite-badge

amadulhaxxani committed:
lint fix

lint fix

amadulhaxxani committed:
Downgrade @types/d3-dispatch to version 3.0.2

Changed the @types/d3-dispatch dependency from 3.0.6 to 3.0.2 in package.json. This may address compatibility or stability issues with the newer version.

kosarko committed:
Merge branch 'clarin-v7' into 58-datacite-badge

kosarko committed:
Merge branch 'clarin-v7' into 58-datacite-badge

amadulhaxxani committed:
Merge branch 'clarin-v7' into 58-datacite-badge

amadulhaxxani committed:
Dedupe @types/d3-dispatch in yarn.lock

Consolidate @types/d3-dispatch entries in yarn.lock to version 3.0.6 and remove the older 3.0.2 entry. This normalizes dependency resolution (likely from running install/upgrade) and contains no source code changes.
